### PR TITLE
Fix our documentation until we release Reek 5.

### LINKED ISSUE #1327: Documentation: default config given in README has the wrong name.
In the README, it says that the default name for the config file is ".reek.yml", but apparently it's anything ending in ".reek".

### PR DESCRIPTION
**Pull request by troessner**
Fixes #1327

### CODE REVIEW

**Comment by mvz** [comment]
Oh man I'm torn about this.

I was going to say if we're going for the solution where we have to change the README just before the release we can just put a pointer to the v4.8.1 README, so documentation an code actually match for the given repo state.

However, this PR also solves the problem that all our `-U` links link to the master branch. The really terrible thing is that even after we release 5.0 there will be people using 4.x and getting broken links.

**Comment by troessner** [comment]
>> Putting back the PrimaDonnaMethod documentation, ideally with a notice that it will be renamed in 5.0.

Just did that.

>> Putting just a notice in the README saying 'This is the README for master, please look here for the currently released version'

Unless we make that <marquee> and <h1> i doubt people are reading that, especially if you quickly want to look something up (what most of our users are doing i suppose) you won't catch that.

File: docs/Rake-Task.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ An more sophisticated rake task that would make use of all available configurati
 ```Ruby
 Reek::Rake::Task.new do |t|
   t.name          = 'custom_rake' # Whatever name you want. Defaults to "reek".
-  t.config_file   = 'config/.reek.yml' # Defaults to nothing.
+  t.config_file   = 'config/config.reek' # Defaults to nothing.
```
Comment (mvz) [inline docs/Rake-Task.md]: This can stay as it is I think.

Comment (troessner) [reply docs/Rake-Task.md]: While it's true that we're not checking for the file extension explicitly when specifying a config file I still think it's very confusing that all of a sudden we have a configuration file with a yaml extension.

**Comment by mvz** [comment]
> Unless we make that "marquee" and "h1" i doubt people are reading that, especially if you quickly want to look something up (what most of our users are doing i suppose) you won't catch that.

I like to think I always notice, but there's no way of knowing, is there? Anyway, let's just merge this so we can think of the Awesomest Best Solution for Pointing People to the Right Documentation (ABSPPRD) at leisure.

**Comment by troessner** [comment]
And merged it is ;)

File: docs/Rake-Task.md (hunk 1 above)
Comment (mvz) [reply docs/Rake-Task.md]: Oh wait I missed that this was also in the docs. Please ignore my remark for this line.

**Pull request merged**